lib/waterline/query/dql/destroy.js

```diff
@@ -74,6 +74,8 @@ module.exports = function(criteria, cb) {
         break;
       }
 
+      var transformedValues = _.map(result, self._transformer.unserialize.bind(self._transformer))
```

Comment:
This makes sense.  The adapters pass back values keyed with column names.  It's the transformer's duty to map those column names to waterline attribute names.  Since the raw adapter method was used [here](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/dql/destroy.js#L55), but [this block of code](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/dql/destroy.js#L66-L75) finds an _attribute_ name, that's why the problem arises.